src/perform_step/adams_bashforth_moulton_perform_step.jl

```diff
@@ -0,0 +1,123 @@
+function initialize!(integrator,cache::AB3ConstantCache)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+@muladd function perform_step!(integrator,cache::AB3ConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  k1 = integrator.fsalfirst
+  cnt = length(integrator.sol.t)
+  if cnt == 1 || cnt == 2
+    ttmp = t + (2/3)*dt
+    ralk2 = f(uprev + (2/3)*dt*k1, p, ttmp)
+    u = uprev + (dt/4)*(k1 + 3*ralk2)
+  else
+    u1 = integrator.sol(t-dt)
```

Comment:
If saving is disabled this won't work. It's better to setup a cache of old values.